burnman/mineral.py

```diff
@@ -64,7 +65,17 @@ class which derives from the equation_of_state base class.
             raise Exception("Please derive your method from object (see python old style classes)")
 
         if self.method is not None and type(new_method) is not type(self.method):
-            warnings.warn('Overriding database equation of state. From ' + self.method.__class__.__name__ + ' to ' + new_method.__class__.__name__, stacklevel=2)
+
+            # Warn user that they are changing the EoS
+            warnings.warn('Overriding database equation of state from ' + self.method.__class__.__name__ + ' to ' + new_method.__class__.__name__ + '. This does not overwrite any mineral attributes other than temperature and pressure. Non-implemented attributes of the old method will remain unchanged unless they are explicitly changed by the user.', stacklevel=2)
+
+            # Set new EoS
+            self.params['equation_of_state']=equation_of_state
```

Comment:
Ok, I see your point, but I wanted to cover the case covered by compare_all_methods. Can we cover both in the warning? (change from MGD3 -> MGD2, originally SLB2), or something similar?

I agree that the param shouldn't be changed, that was not my brightest idea.